### PR TITLE
Problem: authz module is risky

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- it's already disabled in mempool in [v1.0.x](https://github.com/crypto-org-chain/cronos/releases/tag/v1.0.3), now remove completely in v1.1.x.

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

## Summary by CodeRabbit

- **Bug Fixes**
  - Removed deprecated authorization module to streamline permissions and access controls within the app.

- **Refactor**
  - Eliminated unused code related to the authorization module, enhancing app performance and maintainability.

- **Documentation**
  - Updated `CHANGELOG.md` to reflect significant changes in authorization logic.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## UNRELEASED
 
 - [#1258](https://github.com/crypto-org-chain/cronos/pull/1258) Support hard-fork style upgrades.
+- [#]() Remove authz module.
```
Comment (yihuang) [inline CHANGELOG.md:6]:
```suggestion
- [#1274](https://github.com/crypto-org-chain/cronos/pull/1274) Remove authz module.
```

File: app/upgrades.go [hunk 2]
```diff
@@ -9,6 +9,7 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/module"
 	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	authztypes "github.com/cosmos/cosmos-sdk/x/authz/types"
```
Comment (mmsqe) [inline app/upgrades.go:12]: need this [key](https://github.com/cosmos/cosmos-sdk/blob/release/v0.47.x/x/authz/keys.go#L5) right?

Comment (yihuang) [reply app/upgrades.go:12]: yes

**Review by mmsqe — APPROVED** [review]

**Comment by mmsqe** [comment]
seems get wrong Block.Header.AppHash in upgrade test

**Comment by yihuang** [comment]
> seems get wrong Block.Header.AppHash in upgrade test

yeah, maybe some inconsistency between memiavl and iavl handling deletion of store.

**Review by mmsqe — APPROVED** [review]

**Pull request merged**